packages/zk-transfer-web/src/views/MakeTransaction.tsx

```diff
@@ -29,10 +30,30 @@ const MakeTransaction = () => {
     ["MKR", "#AF005E", 500, 500]
   ];
 
+  const generateUsername = () => {
+    const regex = /(\w+)_(\w+)/gi;
+    const capitalize = (str: string) => str[0].toUpperCase() + str.slice(1);
+    const replaceFn = (tokn: any, $1: any, $2: any) => capitalize($1) + " " + capitalize($2);
+    return usernameGenerator.getRandomName().replace(regex, replaceFn);
+  }
+
+  const demo: any = useContext(Democracy);
+  let username: any = '';
+
+  if (demo.chainId) {
```

Comment:
How does this interact with the file `src/App.tsx` and can
you combine it with these lines there to simplify the calls to `useState`?
```
function App({ demo } : AppProp) {
  const [chainId, setDemoChain] = useState(demo.chainId);

  useEffect(() => {
    if (!chainId) {
      demo.clientInit()
        .then(() => setDemoChain(demo.chainId));
    }   
  }); 
```

Reply:
Hey i have moved the logic one level up(App.tsx) so that the screename can be accessed by other components.